src/unity/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -202,7 +202,7 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
         _tkutl._raise_error_if_sframe_empty(validation_set, 'validation_set')
         validation_set = _tkutl._toolkits_select_columns(
             validation_set, features + [session_id, target])
-        validation_set = validation_set.filter_by(target_map.keys(), target)
+        validation_set = validation_set.filter_by(list(target_map.keys()), target)
```

Comment:
Hmm. This fix is fine for now, but I wonder if `filter_by` should accept any iterable (not just a list) which would allow the code to work as is?

Reply:
It turns out that this boils down to the fact that you cannot pass dict_keys to the SArray initializer:

```
>>> dict = {'alpha': 0, 'beta':1}
>>> sa = tc.SArray(dict.keys())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jong/Source/turicreate/debug/src/unity/python/turicreate/data_structures/sarray.py", line 422, in __init__
    raise TypeError("Unexpected data source. " \
TypeError: Unexpected data source. Possible data source types are: list, numpy.ndarray, pandas.Series, and string(url)
```

Yeah, not ready to change SArray.__init__ now, but is this worth tracking?

Reply:
Yeah, I think so. Let's open another issue for it. I think `init` should allow any iterable for this parameter.

Reply:
Sounds good, filed #915